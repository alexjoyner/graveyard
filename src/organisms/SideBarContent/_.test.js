import React from 'react';
import { shallow } from 'enzyme';
import { BaseSideBarContent } from './';

describe('BaseSideBarContent component', () => {
  let component;
  beforeEach(() => {
    const props = {
      changePointGroup: jest.fn(),
      groups: [
        {id: 1, name: 'test'}, 
        {id: 2, name: 'test2'},
      ],
    }
    component = shallow(<BaseSideBarContent {...props} />);
  });
  it('Should render without exploding', () => {
    expect(component).toBeDefined();
  });
  it('Should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
