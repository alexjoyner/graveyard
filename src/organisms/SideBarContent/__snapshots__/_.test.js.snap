// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSideBarContent component Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BaseSideBarContent
    changePointGroup={[MockFunction]}
    groups={
      Array [
        Object {
          "id": 1,
          "name": "test",
        },
        Object {
          "id": 2,
          "name": "test2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Groups
        </ForwardRef>,
        <ForwardRef>
          <ListItem
            onClick={[Function]}
          >
            test
          </ListItem>
          <ListItem
            onClick={[Function]}
          >
            test2
          </ListItem>
        </ForwardRef>,
        <Connect(BaseMultiSelectedChartsMenu) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Groups",
        },
        "ref": null,
        "rendered": "Groups",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-jKJlTe",
            "isStatic": false,
            "rules": Array [
              "
  -moz-box-sizing: border-box; 
  -webkit-box-sizing: border-box; 
  box-sizing: border-box; 
  top: 0;
  left: 0;
  background-color: ",
              [Function],
              ";
  border: none;
  width: 100%;
  height: 41px;
  line-height: 41px;
  vertical-align: middle;
  padding-left: 10px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-jKJlTe",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ListItem
              onClick={[Function]}
            >
              test
            </ListItem>,
            <ListItem
              onClick={[Function]}
            >
              test2
            </ListItem>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "children": "test",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "children": "test2",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "test2",
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-dxgOiQ",
            "isStatic": true,
            "rules": Array [
              "
    text-align: center;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-dxgOiQ",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Groups
          </ForwardRef>,
          <ForwardRef>
            <ListItem
              onClick={[Function]}
            >
              test
            </ListItem>
            <ListItem
              onClick={[Function]}
            >
              test2
            </ListItem>
          </ForwardRef>,
          <Connect(BaseMultiSelectedChartsMenu) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Groups",
          },
          "ref": null,
          "rendered": "Groups",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-jKJlTe",
              "isStatic": false,
              "rules": Array [
                "
  -moz-box-sizing: border-box; 
  -webkit-box-sizing: border-box; 
  box-sizing: border-box; 
  top: 0;
  left: 0;
  background-color: ",
                [Function],
                ";
  border: none;
  width: 100%;
  height: 41px;
  line-height: 41px;
  vertical-align: middle;
  padding-left: 10px;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-jKJlTe",
            "target": "div",
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ListItem
                onClick={[Function]}
              >
                test
              </ListItem>,
              <ListItem
                onClick={[Function]}
              >
                test2
              </ListItem>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "children": "test",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "test",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "children": "test2",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "test2",
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-dxgOiQ",
              "isStatic": true,
              "rules": Array [
                "
    text-align: center;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-dxgOiQ",
            "target": "div",
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
